custom_components/nefiteasy/switch.py

```diff
@@ -143,3 +147,103 @@ def get_endpoint(self) -> str:
             else "dhwOperationManualMode"
         )
         return "/dhwCircuits/dhwA/" + endpoint
+
+
+class NefitHolidayMode(NefitSwitch):
+    """Class for nefit holiday mode entity.
+
+       Note it only works if the thermostat preset is set to Clock
+    """
+
+    climate_entity = None
+    climate_preset = None
+    config_entry   = None
+
+    def __init__(
+        self,
+        entity_description: NefitEntityDescription,
+        client: NefitEasy,
+        data: MappingProxyType[str, Any],
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(entity_description, client, data)
+
+        self.config_entry = config_entry
+
+    def get_climate_entity( self ):
+        if self.climate_entity is None:
+            self.climate_entity = self.hass.data[DOMAIN][self.config_entry.entry_id][ "climate" ]
+
+        return self.climate_entity
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+
+        # Save and current preset mode and change it to Clock
+        self.climate_preset = self.get_climate_entity().preset_mode
+
+        _LOGGER.debug(
+            "Forcing thermostat preset mode to %s, previous mode %s.",
+            OPERATION_CLOCK,
+            self.climate_preset,
+        )
+
+        await self.get_climate_entity().async_set_preset_mode(OPERATION_CLOCK)
```

Comment:
Can we directly send the operation mode to the device without using the climate entity?